src/jitlayers.cpp

```diff
@@ -1070,6 +1070,7 @@ void jl_dump_native(const char *bc_fname, const char *unopt_bc_fname, const char
     TheTriple.setObjectFormat(Triple::COFF);
 #elif defined(_OS_DARWIN_)
     TheTriple.setObjectFormat(Triple::MachO);
+    TheTriple.setOS(llvm::Triple::MacOSX);
```

Comment:
In the off chance that Julia works on PureDarwin (I assume no one has ever tried), perhaps this should be in its own `#ifdef __APPLE__`?